### PR TITLE
Add non-secure configuration for Thingy53 samples

### PR DESCRIPTION
**Pull request by emob-nordic**
The configuration is a copy-paste from secure side, the only difference is:
- files use `_ns` suffix
- `pm_static_*.yml` adds spm partition

I've tested the samples by uploading the images in serial recovery mode.

### CODE REVIEW

File: samples/bluetooth/peripheral_uart/sample.yaml
Comment (KAGA164) [inline samples/bluetooth/peripheral_uart/sample.yaml:33]: Why did you remove `thingy53_nrf5340_cpuapp` ? Wasn't cdc add for Thingy purpose ?

Comment (emob-nordic) [reply samples/bluetooth/peripheral_uart/sample.yaml:33]: Glad you asked. That was a mistake and should not be added. CDC is indeed enabled in Thingy:53 configurations. Thingy:53 uses 2 CDC_ACM instances, one for logging and one acting as UART backend for the sample purposes.

This overlay was added to test build configurations for other boards with async adapter. Thingy:53 has this enabled in the default configuration.

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:1]: Matter weather station is missing. We could add it or contact maintainers regarding this one.

File: samples/bluetooth/mesh/light/CMakeLists.txt
Comment (MarekPieta) [inline samples/bluetooth/mesh/light/CMakeLists.txt:25]: Maybe
```
if (CONFIG_BOARD_THINGY53_NRF5340_CPUAPP OR CONFIG_BOARD_THINGY53_NRF5340_CPUAPP_NS)
   target_sources...
endif()
```
Also for other samples

Comment (MarekPieta) [reply samples/bluetooth/mesh/light/CMakeLists.txt:25]: It's already done like this for board related files in Zephyr

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:14]: Please update the following values in app configurations:
```
CONFIG_BT_L2CAP_TX_MTU=498
CONFIG_BT_BUF_ACL_RX_SIZE=502
```

Comment (MarekPieta) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:14]: Also make sure to update the following one in net core configuration
`CONFIG_BT_BUF_ACL_RX_SIZE=502`

Comment (MarekPieta) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:14]: Needed to speed up DFU

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:1]: I already discussed that and they don't plan to add ns support so i left this.

Comment (pdunaj) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:1]: To be honest we don't have to do it either. The goal was to make sure it's possible.
I would still add the configuration to machine learning or some samples so we are sure this thing keeps building.

File: samples/bluetooth/mesh/light/CMakeLists.txt
Comment (pdunaj) [reply samples/bluetooth/mesh/light/CMakeLists.txt:25]: I don't mind it being as it is now. It's clear to distinguish board support and provides little help for anything else.

**Review by pdunaj — APPROVED** [review]

**Comment by pdunaj** [comment]
Two questions - are the secure non-secure apps swappable on the same bootloader?

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:14]: Shouldn't we make it in different PR? Its just about adding the configuration. We can create separate PR for speeding up the dfu.

**Comment by emob-nordic** [comment]
> Are the secure non-secure apps swappable on the same bootloader?

Checked once again to be sure. Yes they are.

**Comment by emob-nordic** [comment]
Last push is for changing:
```
CONFIG_BT_L2CAP_TX_MTU=498
CONFIG_BT_BUF_ACL_RX_SIZE=502
```
as suggested by @MarekPieta

**Comment by emob-nordic** [comment]
Last push is a rebase due to weird compliance and doc build fails.

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf
Comment (MarekPieta) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp_ns/app_ZDebug.conf:14]: The DFU speed up was already added for cpuapp, so it should be added here too, Discussed offline.

**Review by MarekPieta — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Review by rakons — CHANGES_REQUESTED** [review]
Please add an information to changelog.

**Comment by emob-nordic** [comment]
> Please add an information to changelog.

done.

**Comment by emob-nordic** [comment]
Rebased due to conflicting file.

File: doc/nrf/releases/release-notes-changelog.rst
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:103]:
```suggestion
  * Updated some samples with support for :ref:`zephyr:thingy53_nrf5340` in non-secure configuration.
```

File: doc/nrf/releases/release-notes-changelog.rst
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:110]:
```suggestion
  * Updated some samples with support for :ref:`zephyr:thingy53_nrf5340` in non-secure configuration.
```

File: doc/nrf/releases/release-notes-changelog.rst
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:132]:
```suggestion
* Added non-secure configuration for building with :ref:`zephyr:thingy53_nrf5340`.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
A couple of little things that caught my eye.

**Comment by emob-nordic** [comment]
Pushed changes requested by @wiba-nordic

**Review by wiba-nordic — APPROVED** [review]
Looks good.

**Review by pdunaj — APPROVED** [review]

**Comment by pdunaj** [comment]
@emob-nordic , @rakons , can we merge this one?

**Comment by emob-nordic** [comment]
> @emob-nordic , @rakons , can we merge this one?

Well, from my pov, yes.

File: doc/nrf/releases/release-notes-changelog.rst
Comment (rakons) [inline doc/nrf/releases/release-notes-changelog.rst:103]: * Updated ~~some~~ samples with support for...

File: doc/nrf/releases/release-notes-changelog.rst
Comment (rakons) [inline doc/nrf/releases/release-notes-changelog.rst:110]: * Updated ~~some~~ samples with support for...

**Review by rakons — APPROVED** [review]
Small style comment but it may also by handled by documentation team.

File: doc/nrf/releases/release-notes-changelog.rst
Comment (wiba-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:103]: That "some" was actually added by me. From what I could tell, not all Bluetooth LE samples were updated.

File: doc/nrf/releases/release-notes-changelog.rst
Comment (emob-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:110]: I added `some` on @wiba-nordic  request :(

File: doc/nrf/releases/release-notes-changelog.rst
Comment (rakons) [reply doc/nrf/releases/release-notes-changelog.rst:103]: Well... it is just a comment. I really do not like "some" in a documentation and the rest of the comment explains with samples configuration was added. I am not blocking merge anyway - As I marked in my approval - if documentation team agrees that I have nothing to add.

File: doc/nrf/releases/release-notes-changelog.rst
Comment (rakons) [reply doc/nrf/releases/release-notes-changelog.rst:110]: Just a comment - not a blocker as I stated in previous comment.

Comment (wiba-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:110]: I'll ask the rest of the doc team for opinions.

**Pull request merged**
